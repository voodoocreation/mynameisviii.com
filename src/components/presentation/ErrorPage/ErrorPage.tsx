import Head from "next/head";
import * as React from "react";
import { MdErrorOutline } from "react-icons/md";
import { InjectedIntlProps, injectIntl } from "react-intl";

import { absoluteUrl } from "../../../helpers/dataTransformers";
import PageHeader from "../../presentation/PageHeader/PageHeader";

import "./ErrorPage.scss";

interface IProps extends InjectedIntlProps {
  message?: string;
  status?: number;
}

class ErrorPage extends React.Component<IProps> {
  public static defaultProps = {
    status: 500
  };

  public render() {
    const { formatMessage } = this.props.intl;

    return (
      <React.Fragment>
        <Head>
          <title>
            {`${this.getTitle()} Â· ${formatMessage({
              id: "BRAND_NAME"
            })}`}
          </title>

          <meta content={this.getMessage()} name="description" />
          <meta property="og:title" content={this.getTitle()} />
          <meta property="og:description" content={this.getMessage()} />
          <meta property="og:type" content="website" />
          <meta
            property="og:image"
            content="https://s3.amazonaws.com/mynameisviii-static/theme/heart/og/error.jpg"
          />
          <meta property="og:url" content={absoluteUrl("/symbol")} />
        </Head>

        <article className="ErrorPage">
          <PageHeader>{this.getTitle()}</PageHeader>

          <div className="ErrorPage--content">
            <MdErrorOutline />
            <p>{this.getMessage()}</p>
          </div>
        </article>
      </React.Fragment>
    );
  }

  private getTitle() {
    const { status } = this.props;
    const { formatMessage } = this.props.intl;

    switch (status) {
      default:
        return formatMessage({ id: "ERROR_TITLE" });

      case 404:
        return formatMessage({ id: "ERROR_404_TITLE" });
    }
  }

  private getMessage() {
    const { message, status } = this.props;
    const { formatMessage } = this.props.intl;

    switch (status) {
      default:
        return message ? message : formatMessage({ id: "ERROR_MESSAGE" });

      case 404:
        return formatMessage({ id: "ERROR_404_MESSAGE" });
    }
  }
}

export default injectIntl(ErrorPage);
