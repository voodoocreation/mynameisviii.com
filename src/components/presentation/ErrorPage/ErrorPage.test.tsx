import messages from "../../../locales/en-NZ";
import WrapperWithIntl from "../../../utilities/WrapperWithIntl";
import ErrorPage from "./ErrorPage";

const component = new WrapperWithIntl(ErrorPage);

describe("[presentation] <ErrorPage />", () => {
  describe("when status is 404", () => {
    const wrapper = component
      .withProps({
        status: 404,
      })
      .render();

    it("renders the correct title", () => {
      expect(wrapper.find(".PageHeader").text()).toBe(messages.ERROR_404_TITLE);
    });

    it("renders the correct message", () => {
      expect(wrapper.find(".ErrorPage--content").text()).toBe(
        messages.ERROR_404_MESSAGE
      );
    });

    it("matches snapshot", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("when status is 500", () => {
    const wrapper = component
      .withProps({
        status: 500,
      })
      .render();

    it("renders the correct title", () => {
      expect(wrapper.find(".PageHeader").text()).toBe(messages.ERROR_TITLE);
    });

    it("renders the correct message", () => {
      expect(wrapper.find(".ErrorPage--content").text()).toBe(
        messages.ERROR_MESSAGE
      );
    });

    it("matches snapshot", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("when message is defined", () => {
    const message = "Test message";

    const wrapper = component
      .withProps({
        message,
      })
      .render();

    it("renders the correct message", () => {
      expect(wrapper.find(".ErrorPage--content").text()).toBe(message);
    });

    it("matches snapshot", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
