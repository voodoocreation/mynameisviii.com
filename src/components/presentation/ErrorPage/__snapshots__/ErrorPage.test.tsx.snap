// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[presentation] <ErrorPage /> when message is defined matches snapshot 1`] = `
<article
  class="ErrorPage"
>
  <header
    class="PageHeader"
  >
    <h1>
      An error has occurred
    </h1>
  </header>
  <div
    class="ErrorPage--content"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
      />
    </svg>
    <p>
      Test message
    </p>
  </div>
</article>
`;

exports[`[presentation] <ErrorPage /> when status is 404 matches snapshot 1`] = `
<article
  class="ErrorPage"
>
  <header
    class="PageHeader"
  >
    <h1>
      Page not found
    </h1>
  </header>
  <div
    class="ErrorPage--content"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
      />
    </svg>
    <p>
      This page does not exist - please use the site navigation to find what you're looking for.
    </p>
  </div>
</article>
`;

exports[`[presentation] <ErrorPage /> when status is 500 matches snapshot 1`] = `
<article
  class="ErrorPage"
>
  <header
    class="PageHeader"
  >
    <h1>
      An error has occurred
    </h1>
  </header>
  <div
    class="ErrorPage--content"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
      />
    </svg>
    <p>
      An error has occurred; please try again.
    </p>
  </div>
</article>
`;
